From a3f716e46c53e04a0a45e6d1bbb499c76ddcc674 Mon Sep 17 00:00:00 2001
From: Victor Bo <bvoid@yandex.ru>
Date: Sat, 22 Feb 2020 16:50:09 +0200
Subject: [PATCH 1/2] minimize Roboto fonts set

	modified:   Android.mk
	renamed:    Roboto-Black.ttf -> RobotoMini-Black.ttf
	renamed:    Roboto-BlackItalic.ttf -> RobotoMini-BlackItalic.ttf
	renamed:    Roboto-Bold.ttf -> RobotoMini-Bold.ttf
	renamed:    Roboto-BoldItalic.ttf -> RobotoMini-BoldItalic.ttf
	renamed:    Roboto-Italic.ttf -> RobotoMini-Italic.ttf
	renamed:    Roboto-Light.ttf -> RobotoMini-Light.ttf
	renamed:    Roboto-LightItalic.ttf -> RobotoMini-LightItalic.ttf
	renamed:    Roboto-Medium.ttf -> RobotoMini-Medium.ttf
	renamed:    Roboto-MediumItalic.ttf -> RobotoMini-MediumItalic.ttf
	renamed:    Roboto-Regular.ttf -> RobotoMini-Regular.ttf
	renamed:    Roboto-Thin.ttf -> RobotoMini-Thin.ttf
	renamed:    Roboto-ThinItalic.ttf -> RobotoMini-ThinItalic.ttf
	renamed:    RobotoCondensed-Bold.ttf -> RobotoMiniCondensed-Bold.ttf
	renamed:    RobotoCondensed-BoldItalic.ttf -> RobotoMiniCondensed-BoldItalic.ttf
	renamed:    RobotoCondensed-Italic.ttf -> RobotoMiniCondensed-Italic.ttf
	renamed:    RobotoCondensed-Light.ttf -> RobotoMiniCondensed-Light.ttf
	renamed:    RobotoCondensed-LightItalic.ttf -> RobotoMiniCondensed-LightItalic.ttf
	renamed:    RobotoCondensed-Medium.ttf -> RobotoMiniCondensed-Medium.ttf
	renamed:    RobotoCondensed-MediumItalic.ttf -> RobotoMiniCondensed-MediumItalic.ttf
	renamed:    RobotoCondensed-Regular.ttf -> RobotoMiniCondensed-Regular.ttf
	modified:   fonts.mk
---
 Android.mk                                    |  68 ++----------------
 Roboto-Black.ttf => RobotoMini-Black.ttf      | Bin
 ...ckItalic.ttf => RobotoMini-BlackItalic.ttf | Bin
 Roboto-Bold.ttf => RobotoMini-Bold.ttf        | Bin
 ...oldItalic.ttf => RobotoMini-BoldItalic.ttf | Bin
 Roboto-Italic.ttf => RobotoMini-Italic.ttf    | Bin
 Roboto-Light.ttf => RobotoMini-Light.ttf      | Bin
 ...htItalic.ttf => RobotoMini-LightItalic.ttf | Bin
 Roboto-Medium.ttf => RobotoMini-Medium.ttf    | Bin
 ...mItalic.ttf => RobotoMini-MediumItalic.ttf | Bin
 Roboto-Regular.ttf => RobotoMini-Regular.ttf  | Bin
 Roboto-Thin.ttf => RobotoMini-Thin.ttf        | Bin
 ...hinItalic.ttf => RobotoMini-ThinItalic.ttf | Bin
 ...d-Bold.ttf => RobotoMiniCondensed-Bold.ttf | Bin
 ....ttf => RobotoMiniCondensed-BoldItalic.ttf | Bin
 ...alic.ttf => RobotoMiniCondensed-Italic.ttf | Bin
 ...Light.ttf => RobotoMiniCondensed-Light.ttf | Bin
 ...ttf => RobotoMiniCondensed-LightItalic.ttf | Bin
 ...dium.ttf => RobotoMiniCondensed-Medium.ttf | Bin
 ...tf => RobotoMiniCondensed-MediumItalic.ttf | Bin
 ...lar.ttf => RobotoMiniCondensed-Regular.ttf | Bin
 fonts.mk                                      |  26 ++-----
 22 files changed, 12 insertions(+), 82 deletions(-)
 rename Roboto-Black.ttf => RobotoMini-Black.ttf (100%)
 rename Roboto-BlackItalic.ttf => RobotoMini-BlackItalic.ttf (100%)
 rename Roboto-Bold.ttf => RobotoMini-Bold.ttf (100%)
 rename Roboto-BoldItalic.ttf => RobotoMini-BoldItalic.ttf (100%)
 rename Roboto-Italic.ttf => RobotoMini-Italic.ttf (100%)
 rename Roboto-Light.ttf => RobotoMini-Light.ttf (100%)
 rename Roboto-LightItalic.ttf => RobotoMini-LightItalic.ttf (100%)
 rename Roboto-Medium.ttf => RobotoMini-Medium.ttf (100%)
 rename Roboto-MediumItalic.ttf => RobotoMini-MediumItalic.ttf (100%)
 rename Roboto-Regular.ttf => RobotoMini-Regular.ttf (100%)
 rename Roboto-Thin.ttf => RobotoMini-Thin.ttf (100%)
 rename Roboto-ThinItalic.ttf => RobotoMini-ThinItalic.ttf (100%)
 rename RobotoCondensed-Bold.ttf => RobotoMiniCondensed-Bold.ttf (100%)
 rename RobotoCondensed-BoldItalic.ttf => RobotoMiniCondensed-BoldItalic.ttf (100%)
 rename RobotoCondensed-Italic.ttf => RobotoMiniCondensed-Italic.ttf (100%)
 rename RobotoCondensed-Light.ttf => RobotoMiniCondensed-Light.ttf (100%)
 rename RobotoCondensed-LightItalic.ttf => RobotoMiniCondensed-LightItalic.ttf (100%)
 rename RobotoCondensed-Medium.ttf => RobotoMiniCondensed-Medium.ttf (100%)
 rename RobotoCondensed-MediumItalic.ttf => RobotoMiniCondensed-MediumItalic.ttf (100%)
 rename RobotoCondensed-Regular.ttf => RobotoMiniCondensed-Regular.ttf (100%)

diff --git a/Android.mk b/Android.mk
index 8094048..77ce5d0 100644
--- a/Android.mk
+++ b/Android.mk
@@ -12,26 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# We have to use BUILD_PREBUILT instead of PRODUCT_COPY_FIES,
-# because MINIMAL_FONT_FOOTPRINT is only available in Android.mks.
-
 LOCAL_PATH := $(call my-dir)
 
-##########################################
-# create symlink for given font
-# $(1): new font $(2): link target
-# should be used with eval: $(eval $(call ...))
-define create-font-symlink
-$(PRODUCT_OUT)/system/fonts/$(1) : $(PRODUCT_OUT)/system/fonts/$(2)
-	@echo "Symlink: $$@ -> $$<"
-	@mkdir -p $$(dir $$@)
-	@rm -rf $$@
-	$(hide) ln -sf $$(notdir $$<) $$@
-# this magic makes LOCAL_REQUIRED_MODULES work
-ALL_MODULES.$(1).INSTALLED := \
-    $(ALL_MODULES.$(1).INSTALLED) $(PRODUCT_OUT)/system/fonts/$(1)
-endef
-
 # Build the rest of font files as prebuilt.
 # $(1): The source file name in LOCAL_PATH.
 #       It also serves as the module name and the dest file name.
@@ -46,50 +28,12 @@ $(eval include $(BUILD_PREBUILT))
 endef
 
 font_src_files := \
-    Roboto-Regular.ttf \
-    Roboto-Bold.ttf \
-    Roboto-Italic.ttf \
-    Roboto-BoldItalic.ttf
-
-ifeq ($(MINIMAL_FONT_FOOTPRINT),true)
-
-$(eval $(call create-font-symlink,Roboto-Black.ttf,Roboto-Bold.ttf))
-$(eval $(call create-font-symlink,Roboto-BlackItalic.ttf,Roboto-BoldItalic.ttf))
-$(eval $(call create-font-symlink,Roboto-Light.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,Roboto-LightItalic.ttf,Roboto-Italic.ttf))
-$(eval $(call create-font-symlink,Roboto-Medium.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,Roboto-MediumItalic.ttf,Roboto-Italic.ttf))
-$(eval $(call create-font-symlink,Roboto-Thin.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,Roboto-ThinItalic.ttf,Roboto-Italic.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-Regular.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-Bold.ttf,Roboto-Bold.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-Italic.ttf,Roboto-Italic.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-BoldItalic.ttf,Roboto-BoldItalic.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-Light.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-LightItalic.ttf,Roboto-Italic.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-Medium.ttf,Roboto-Regular.ttf))
-$(eval $(call create-font-symlink,RobotoCondensed-MediumItalic.ttf,Roboto-Italic.ttf))
-
-else # !MINIMAL_FONT
-font_src_files += \
-    Roboto-Black.ttf \
-    Roboto-BlackItalic.ttf \
-    Roboto-Light.ttf \
-    Roboto-LightItalic.ttf \
-    Roboto-Medium.ttf \
-    Roboto-MediumItalic.ttf \
-    Roboto-Thin.ttf \
-    Roboto-ThinItalic.ttf \
-    RobotoCondensed-Regular.ttf \
-    RobotoCondensed-Bold.ttf \
-    RobotoCondensed-Italic.ttf \
-    RobotoCondensed-BoldItalic.ttf \
-    RobotoCondensed-Light.ttf \
-    RobotoCondensed-LightItalic.ttf \
-    RobotoCondensed-Medium.ttf \
-    RobotoCondensed-MediumItalic.ttf
-
-endif # !MINIMAL_FONT
+    RobotoMini-Regular.ttf \
+    RobotoMini-Bold.ttf \
+    RobotoMini-Italic.ttf \
+    RobotoMini-BoldItalic.ttf \
+    RobotoMini-Medium.ttf \
+    RobotoMini-MediumItalic.ttf
 
 $(foreach f, $(font_src_files), $(call build-one-font-module, $(f)))
 
diff --git a/Roboto-Black.ttf b/RobotoMini-Black.ttf
similarity index 100%
rename from Roboto-Black.ttf
rename to RobotoMini-Black.ttf
diff --git a/Roboto-BlackItalic.ttf b/RobotoMini-BlackItalic.ttf
similarity index 100%
rename from Roboto-BlackItalic.ttf
rename to RobotoMini-BlackItalic.ttf
diff --git a/Roboto-Bold.ttf b/RobotoMini-Bold.ttf
similarity index 100%
rename from Roboto-Bold.ttf
rename to RobotoMini-Bold.ttf
diff --git a/Roboto-BoldItalic.ttf b/RobotoMini-BoldItalic.ttf
similarity index 100%
rename from Roboto-BoldItalic.ttf
rename to RobotoMini-BoldItalic.ttf
diff --git a/Roboto-Italic.ttf b/RobotoMini-Italic.ttf
similarity index 100%
rename from Roboto-Italic.ttf
rename to RobotoMini-Italic.ttf
diff --git a/Roboto-Light.ttf b/RobotoMini-Light.ttf
similarity index 100%
rename from Roboto-Light.ttf
rename to RobotoMini-Light.ttf
diff --git a/Roboto-LightItalic.ttf b/RobotoMini-LightItalic.ttf
similarity index 100%
rename from Roboto-LightItalic.ttf
rename to RobotoMini-LightItalic.ttf
diff --git a/Roboto-Medium.ttf b/RobotoMini-Medium.ttf
similarity index 100%
rename from Roboto-Medium.ttf
rename to RobotoMini-Medium.ttf
diff --git a/Roboto-MediumItalic.ttf b/RobotoMini-MediumItalic.ttf
similarity index 100%
rename from Roboto-MediumItalic.ttf
rename to RobotoMini-MediumItalic.ttf
diff --git a/Roboto-Regular.ttf b/RobotoMini-Regular.ttf
similarity index 100%
rename from Roboto-Regular.ttf
rename to RobotoMini-Regular.ttf
diff --git a/Roboto-Thin.ttf b/RobotoMini-Thin.ttf
similarity index 100%
rename from Roboto-Thin.ttf
rename to RobotoMini-Thin.ttf
diff --git a/Roboto-ThinItalic.ttf b/RobotoMini-ThinItalic.ttf
similarity index 100%
rename from Roboto-ThinItalic.ttf
rename to RobotoMini-ThinItalic.ttf
diff --git a/RobotoCondensed-Bold.ttf b/RobotoMiniCondensed-Bold.ttf
similarity index 100%
rename from RobotoCondensed-Bold.ttf
rename to RobotoMiniCondensed-Bold.ttf
diff --git a/RobotoCondensed-BoldItalic.ttf b/RobotoMiniCondensed-BoldItalic.ttf
similarity index 100%
rename from RobotoCondensed-BoldItalic.ttf
rename to RobotoMiniCondensed-BoldItalic.ttf
diff --git a/RobotoCondensed-Italic.ttf b/RobotoMiniCondensed-Italic.ttf
similarity index 100%
rename from RobotoCondensed-Italic.ttf
rename to RobotoMiniCondensed-Italic.ttf
diff --git a/RobotoCondensed-Light.ttf b/RobotoMiniCondensed-Light.ttf
similarity index 100%
rename from RobotoCondensed-Light.ttf
rename to RobotoMiniCondensed-Light.ttf
diff --git a/RobotoCondensed-LightItalic.ttf b/RobotoMiniCondensed-LightItalic.ttf
similarity index 100%
rename from RobotoCondensed-LightItalic.ttf
rename to RobotoMiniCondensed-LightItalic.ttf
diff --git a/RobotoCondensed-Medium.ttf b/RobotoMiniCondensed-Medium.ttf
similarity index 100%
rename from RobotoCondensed-Medium.ttf
rename to RobotoMiniCondensed-Medium.ttf
diff --git a/RobotoCondensed-MediumItalic.ttf b/RobotoMiniCondensed-MediumItalic.ttf
similarity index 100%
rename from RobotoCondensed-MediumItalic.ttf
rename to RobotoMiniCondensed-MediumItalic.ttf
diff --git a/RobotoCondensed-Regular.ttf b/RobotoMiniCondensed-Regular.ttf
similarity index 100%
rename from RobotoCondensed-Regular.ttf
rename to RobotoMiniCondensed-Regular.ttf
diff --git a/fonts.mk b/fonts.mk
index 2ba713e..3d27fcd 100644
--- a/fonts.mk
+++ b/fonts.mk
@@ -13,23 +13,9 @@
 # limitations under the License.
 
 PRODUCT_PACKAGES := \
-    Roboto-Regular.ttf \
-    Roboto-Bold.ttf \
-    Roboto-Italic.ttf \
-    Roboto-BoldItalic.ttf \
-    Roboto-Black.ttf \
-    Roboto-BlackItalic.ttf \
-    Roboto-Light.ttf \
-    Roboto-LightItalic.ttf \
-    Roboto-Medium.ttf \
-    Roboto-MediumItalic.ttf \
-    Roboto-Thin.ttf \
-    Roboto-ThinItalic.ttf \
-    RobotoCondensed-Regular.ttf \
-    RobotoCondensed-Bold.ttf \
-    RobotoCondensed-Italic.ttf \
-    RobotoCondensed-BoldItalic.ttf \
-    RobotoCondensed-Light.ttf \
-    RobotoCondensed-LightItalic.ttf \
-    RobotoCondensed-Medium.ttf \
-    RobotoCondensed-MediumItalic.ttf
+    RobotoMini-Regular.ttf \
+    RobotoMini-Bold.ttf \
+    RobotoMini-Italic.ttf \
+    RobotoMini-BoldItalic.ttf \
+    RobotoMini-Medium.ttf \
+    RobotoMini-MediumItalic.ttf
-- 
2.25.1

